lib/YuiRestClient/Widget/DebugLabel.pm

```diff
@@ -0,0 +1,24 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+package YuiRestClient::Widget::DebugLabel;
+
+use strict;
+use warnings;
+
+use parent 'YuiRestClient::Widget::Base';
+
+sub text {
+    my ($self, $args) = @_;
+    return $self->property('text');
```

Comment:
I believe that this is the problem you have, I have sles installer running and we have following json:
```
  {
    "class" : "YWizard",
    "debug_label" : "SUSE Linux Enterprise Server 15 SP3 License Agreement",
    "hstretch" : true,
    "id" : "wizard",
    "vstretch" : true
  }
```
So label here doesn't have this property `text`, means that it will return `undef`.

Reply:
Does that mean that in this case return $self->property('debug_label'); would work for me ?

Reply:
In my understanding, if I had here $self->property('debug_label'), I would get the text "SUSE Linux Enterprise Server 15 SP3 License Agreement" returned ? That's how I first undertood it, so this would allow to retrieve and compare that text, and could even be sent to a record_info

Reply:
Depends on what you have as `$self`, but yes, it should be it. Also, keep in mind that it doesn't make sense to assert label once we have waited for it, so checking that it is displayed should be enough here.

Reply:
So, if I understand correcly, $self should be here {app}->label({'debug_label' => 'Some text'});. (?) So I tried to write $self->property('debug_label'), but it still seem to return undef, and therefore does not fail nor return anything, even when the text is right.

Reply:
So if I understand well property searches in the first array of the json, but here it does not work because what I want is in the second array: 
curl 'http://localhost:9999/v1/widgets?debug_label=Welcome' 
returns:
[
  {
    "class" : "YDialog",
    "hstretch" : true,
    "type" : "wizard",
    "vstretch" : true
  },
  {
    "class" : "YWizard",
    "debug_label" : "Welcome",
    "hstretch" : true,
    "id" : "wizard",
    "vstretch" : true
  },
[...] and more stuff.

So I assumed it would return YDialog here but it seems like it returns 1. At least, I'm now understanding better what to do next. I need to have {app}->label({'class' => 'YDialog'});. and if it returns 1, then it means we're good.

Reply:
So after more examination, the problem is this: 'http://localhost:9999/v1/widgets?debug_label=WHATEVER NON_EXISTING TEXT' also returns the same exact json. So in fact it does not always return undef, but always returns 1 ! So there is no way to assert a page with this. We could grep the whole json output for a that hash, though.